File: signature_demo/php/signature.php

<?php

function milliseconds() {
    list($msec, $sec) = explode(' ', microtime());
    return (int) ($sec . substr($msec, 2, 3));
}

function hmac($msg, $secret) {
    $hmac = hash_hmac('sha256', $msg, $secret, true);
    $hmac = base64_encode($hmac);
    return $hmac;
}

$secret = '98dRnVqbIhriRMZJfax3EvSPRBJrIuZ6J755KcLz6napGUpzmApJmfBY7EY3yYZn';
$api_path = 'user/info';
//$timestamp = (string) milliseconds();
$timestamp = '1562952827927000'; // Please use the timestamp (in miliseconds) in request head 
$msg = $timestamp . '+' . $api_path;
$signature = hmac($msg, $secret);
echo 'prehash msg: '.$msg.PHP_EOL;
echo 'signature: '.$signature.PHP_EOL;

File: signature_demo/cpp/signature.cpp
#include "ascendex_util.h"


int main(){
    // string APIKey = "";
    string secret_key = "hV8FgjyJtpvVeAcMAgzgAFQCN36wmbWuN7o3WPcYcYhFd8qvE43gzFGVsFcCqMNk";
    
    // chrono::milliseconds ms = chrono::duration_cast< chrono::milliseconds >(chrono::system_clock::now().time_since_epoch());
    long timestamp = 1608133910000L; // Please use the same timestamp (in milliseconds) as in request header.

    string msg = to_string(timestamp) + "+info";
    string signature = hmac_sha256(secret_key, msg);

    printf("Prehash msg: %s\n", msg.c_str());
    printf("Signature: %s\n", signature.c_str());


}